src/homework8/arraylist/ArrayList.java

```diff
@@ -0,0 +1,107 @@
+package homework8.arraylist;
+
+import java.util.Arrays;
+
+public class ArrayList<T> implements List<T> {
+    private int position = 0;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[10];
+    }
+
+    @Override
+    public void add(T value) {
+        if (position < array.length) {
```

Comment:
if (position >= array.length) {
            int newArrayLength = array.length + (array.length >> 1);
            array = Arrays.copyOf(array, newArrayLength);
}
            array[position] = value;
            position++;

Reply:
Ok. Thanks for the simplify.